### DISCUSSION TITLE
Feature Request: I can not update redirect node attribute when renderHTML in create note.

### DISCUSSION THREAD

**Discussion by HocNguyenInnomize** [ueberdosis/tiptap#5392] [Feature Requests]
### Description

I use TipTap to create a node with global styles. After creating it and adding various marks, I want to change the styles of the already created nodes. However, I haven’t found a solution to add CSS styles to these nodes. Could you please guide me on how to resolve this issue? (I have used setNodeMarkup and setNodeAttribute, but still cannot add the necessary attribute options to these nodes.)

### Use Case

I use TipTap to create a node with global styles. After creating it and adding various marks, I want to change the styles of the already created nodes. However, I haven’t found a solution to add CSS styles to these nodes. Could you please guide me on how to resolve this issue? (I have used setNodeMarkup and setNodeAttribute, but still cannot add the necessary attribute options to these nodes.)

### Type

New extension

**Comment by nperez0111** [comment]
I would not recommend changing the styles of already existing node (as in adding new CSS after page load for this).

The recommended approach would be to use CSS classnames to change the styling of things. 

It is pretty difficult to know what you are talking about though, could I see some code/design of what you are trying to accomplish?

**Reply by HocNguyenInnomize** [reply]
Thank you for your response. Let me explain again. I have a title in the center position, for example: " h1.. TITLE ...h1", and then I change it to  h6... title ...h6. I want to change the attributes from h1 tag  to  h6 tag while keeping the position unchanged. The same applies to other nodes and HTML tags. Is there a way to do this directly when using renderHTML to create the node? My explanation might be a bit confusing; I hope you understand.

**Reply by nperez0111** [reply]
You can update a node's attributes using the `updateAttributes` method: https://tiptap.dev/docs/editor/api/commands/nodes-and-marks/update-attributes#use-the-updateattributes-command
It will change the node at the current selection to have the new attributes.

renderHTML is not for this at all & should not be dynamic at all.
